pkg/common/mongoutils/mongo.go

```diff
@@ -20,15 +27,32 @@ func GetMongoConnectionStringFromEnv() (string, string, error) {
 	if mongoDBName == "" {
 		return "", "", errors.New("env var 'MONGODB_DATABASE' env var must be set")
 	}
-	if externalConnectionString := os.Getenv("MONGODB_EXTERNAL_CONNECTION_STRING"); externalConnectionString != "" {
+	configdir := os.Getenv("MONGO_CONFIG_DIR")
+
+	if externalConnectionString := getFromEnvOrFile("MONGODB_EXTERNAL_CONNECTION_STRING", configdir+mongoExtCon); externalConnectionString != "" {
 		return externalConnectionString, mongoDBName, nil
 	}
 	mongoDBHost := os.Getenv("MONGODB_HOST")
-	mongoDBUser := os.Getenv("MONGODB_USER")
-	mongoDBPassword := os.Getenv("MONGODB_PASSWORD")
+	mongoDBUser := getFromEnvOrFile("MONGODB_USER", configdir+mongoUser)
+	mongoDBPassword := getFromEnvOrFile("MONGODB_PASSWORD", configdir+mongoPw)
 
 	if !strutils.AllSet(mongoDBHost, mongoDBUser, mongoDBPassword) {
 		return "", "", errors.New("could not construct mongodb connection string: env vars 'MONGODB_HOST', 'MONGODB_USER' and 'MONGODB_PASSWORD' have to be set")
 	}
 	return fmt.Sprintf("mongodb://%s:%s@%s/%s", mongoDBUser, mongoDBPassword, mongoDBHost, mongoDBName), mongoDBName, nil
 }
+
+func readSecret(file string) string {
+	body, err := os.ReadFile(file)
+	if err != nil {
+		logger.Fatalf("unable to read mounted secret: %v", err)
+	}
+	return string(body)
+}
+
+func getFromEnvOrFile(env string, path string) string {
```

Comment:
I would switch the names since we prefer files
```suggestion
func getFromFileOrEnv(env string, path string) string {
```